beacon-chain/rpc/prysm/beacon/handlers.go

```diff
@@ -183,3 +185,48 @@ func (s *Server) GetChainHead(w http.ResponseWriter, r *http.Request) {
 	}
 	httputil.WriteJson(w, response)
 }
+
+func (s *Server) PublishBlobs(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.PublishBlobs")
+	defer span.End()
+	if shared.IsSyncing(r.Context(), w, s.SyncChecker, s.HeadFetcher, s.TimeFetcher, s.OptimisticModeFetcher) {
+		return
+	}
+
+	var req structs.PublishBlobsRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		httputil.HandleError(w, "Could not decode JSON request body", http.StatusBadRequest)
+		return
+	}
+
+	root, err := bytesutil.DecodeHexWithLength(req.BlockRoot, 32)
+	if err != nil {
+		httputil.HandleError(w, "Could not decode block root", http.StatusBadRequest)
+		return
+	}
+
+	for _, blobSidecar := range req.BlobSidecars.Sidecars {
```

Comment:
if BlobSidecars wasn't passed in the request does this panic?

Reply:
Yes.

This payload causes a panic
```
{
        "block_root" : "0x0000000000000000000000000000000000000000000000000000000000000000"
 }
```


This one too
```
fmt.Sprintf(`{
        "block_root" : "0x0000000000000000000000000000000000000000000000000000000000000000",
        "blob_sidecars" : {
           "sidecars" : [
                  {
                         "blob" : "%s",
                         "index" : "0",
                         "kzg_commitment" : "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
                         "kzg_proof" : "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
                         "signed_block_header" : {
                                "signature" : "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
                         }
                  }
           ]
        }
 }`, Blob)
 ```

Seems like missing nil checks on req.BlobSidecars and something with the signed block header.

Reply:
Added more checks and tests